### PR TITLE
Misc fixes

### PR DESCRIPTION
**Pull request by pedromvpg**
This branch bundles UX and reliability improvements across Lightning (including LND REST), ecash, Nostr, backup/developer flows, navigation, and shared UI (tab bar, camera, headers, PIN input). It adds tests where behavior changed and tightens validation for multisig setup.

**Latest updates:** **Bookmark** behavior wired through (beyond filters), **relay connection** handling improvements, **NIP-51** bookmark filters on notes and feed, **NIP-05** and **account cards** on Nostr accounts, **tab bar** gradient typing, and Nostr **account** import/styling cleanup.

## LND REST and Lightning node

- Add **LND REST API connection via connection string**, with improved detection logic and **tests** (`feat` / `fix` commits).
- **Merge** `master` into the LND node QR workstream to stay aligned with upstream.
- Small **LNDRest** border-radius tweak for visual consistency.

## Backup and developer recovery

- Extend **backup recovery** with clearer **state management** for Ark, Lightning, and Nostr identities.
- **Developer recovery**: QR scanning and a more robust **backup decryption** flow.

## Ecash

- Integrate **`SSEcashMintSelector`** for clearer mint selection in ecash transactions.
- Improve handling of **spent proofs** and general **proof management**, including error paths.
- Refine **Ecash account list and detail** layouts and **landing account cards** / balance labels.

## Nostr notes, accounts, and zaps

- Richer **Nostr note** screen: **metadata**, optional **raw JSON** toggle, **NIP-05** validation, **comments**, **zap amounts** with **localized** strings.
- **Bookmarks**: end-to-end **bookmark** behavior plus **NIP-51** bookmark filters on **notes** and **feed** tabs.
- **Relays**: improved **relay connection** handling alongside Nostr flows.
- **NIP-05** validation and improved **account card** display (in addition to NIP-05 on the note screen).
- **Account components**: import cleanup and styling tweaks.
- **Zap sorting** and **payment method** handling improvements (including refactor of shared payment UI).

## Navigation and infrastructure

- **Navigation** helpers updated and covered by **unit tests**.

## Tab bar and shell UI

- **Animated tab bar** with **gradient** treatments, including the **converter** tab.
- **Top border gradients**: stricter **color tuple typing** and clearer gradient definitions (tab shell refactor).

## Camera

- **`SSCameraModal`**: **zoom** and **lens selection** for better capture control.

## Lightning transactions

- **Transaction list/detail**: clearer **status badges**, **raw transaction** visibility where relevant, and cleaner badge logic overall.

## Multisig

- **Simplify multisig key validation** and add a **duplicate xpub** check to avoid invalid setups.

## Cross-cutting UI and fixes

- **Header** button styles aligned across screens.
- **`SSPinInput` / `SSKeyboard`**: behavior and visual consistency; drop unnecessary **`withClear`** usage and align icons.
- **Developer** screen layout/responsiveness.
- Broad **formatting / readability** pass on touched components.

---

## How to test

- **LND**: Connect using REST URL/connection string paths; confirm detection matches expectations (see added tests for edge cases).
- **Recovery**: Walk developer backup recovery with QR and decryption; verify Ark / Lightning / Nostr paths in backup restore.
- **Ecash**: Mint selection, spent-proof flows, account list/detail and landing cards.
- **Nostr**: Open a note—metadata, JSON, NIP-05, comments, zaps (sorting and amounts); exercise **bookmark** add/remove/sync and **NIP-51** filters on notes/feed; watch **relay** behavior under reconnects; confirm **account** cards and **NIP-05** on account routes.
- **Tabs**: Confirm animated gradients and converter tab appearance on devices you care about.
- **Camera**: Zoom and lens switching in `SSCameraModal`.
- **Lightning**: Open transactions—badges and detail/raw sections.
- **Multisig**: Add keys—duplicate xpub blocked; validation feels simpler/clearer.

### CODE REVIEW

**Comment by Psycarlo** [comment]
First, run `pnpm fix` at the root and fix lint/format problems.

File: apps/mobile/api/ecash.ts [hunk 1]
```diff
@@ -234,20 +234,54 @@ export async function createMeltQuote(
...
+}
+
 export async function meltProofs(
   accountId: string,
   mintUrl: string,
   quote: MeltQuote,
   proofs: EcashProof[],
   _description?: string,
-  originalInvoice?: string,
+  _originalInvoice?: string,
```
Comment (Psycarlo) [inline apps/mobile/api/ecash.ts]: If we are not using the argument anymore, remove it

File: apps/mobile/api/ecash.ts [hunk 2]
```diff
@@ -234,20 +234,54 @@ export async function createMeltQuote(
...
+    expiry: quote.expiry,
+    fee_reserve: quote.fee_reserve,
+    payment_preimage: null,
+    quote: quote.quote,
+    request: '',
+    state: 'UNPAID' as const,
+    unit: 'sat'
+  }
+  const result = await wallet.meltProofs(
+    cashuQuote as Parameters<typeof wallet.meltProofs>[0],
```
Comment (Psycarlo) [inline apps/mobile/api/ecash.ts]: Rule 5. Avoid Typescript casting (`as Something`);

File: apps/mobile/api/ecash.ts [hunk 3]
```diff
@@ -234,20 +234,54 @@ export async function createMeltQuote(
...
+    validProofs
+  )
 
   const meltResult = result as unknown as {
```
Comment (Psycarlo) [inline apps/mobile/api/ecash.ts]: Same here

File: apps/mobile/api/nostr.ts [hunk 4]
```diff
@@ -669,6 +773,57 @@ export class NostrAPI {
...
+    await this.connectForPublish()
+    if (!this.ndk) {
+      return new Map()
+    }
+
+    const FETCH_EVENT_BATCH_TIMEOUT_MS = 15000
+    const filter = { ids: validIds, limit: validIds.length }
+    const events = await NostrAPI.fetchManyWithTimeout(
+      this.ndk,
+      filter as Record<string, unknown>,
```
Comment (Psycarlo) [inline apps/mobile/api/nostr.ts]: Avoid `as` cast.

`const filter: Record<string, unknown>` is better

File: apps/mobile/app/(authenticated)/(tabs)/(signer,explorer,converter)/settings/developerRecover.tsx [hunk 5]
```diff
@@ -0,0 +1,302 @@
...
+                    onPress={handleRecoverImSure}
+                    variant="danger"
+                  />
+                </SSVStack>
+              )}
+            </>
+          )}
+        </SSVStack>
+      </ScrollView>
+
```
Comment (Psycarlo) [inline apps/mobile/app/(authenticated)/(tabs)/(signer,explorer,converter)/settings/developerRecover.tsx]: Remove empty line

File: apps/mobile/app/(authenticated)/(tabs)/(signer,explorer,converter)/signer/ecash/account/[id]/settings/index.tsx [hunk 6]
```diff
@@ -81,6 +95,24 @@ export default function EcashAccountSettingsPage() {
...
+              label={t('ecash.account.rename')}
+              onPress={handleRename}
+              variant="outline"
+              disabled={
+                !accountName.trim() ||
+                accountName.trim() === activeAccount?.name
+              }
+            />
+          </SSVStack>
+
```
Comment (Psycarlo) [inline apps/mobile/app/(authenticated)/(tabs)/(signer,explorer,converter)/signer/ecash/account/[id]/settings/index.tsx]: Remove empty line

**Review by Psycarlo — CHANGES_REQUESTED** [review]

File: apps/mobile/app/(authenticated)/(tabs)/(signer,explorer,converter)/settings/developerRecover.tsx [hunk 7]
```diff
@@ -0,0 +1,302 @@
...
+  }
+
+  async function handleRecoverDecrypt() {
+    const raw = recoverEncryptedInput.trim()
+    if (!raw || !recoverPassphrase.trim()) {
+      toast.error(t('settings.developer.backupPassphraseInvalid'))
+      return
+    }
+    try {
+      const payload = JSON.parse(raw) as {
```
Comment (Psycarlo) [inline apps/mobile/app/(authenticated)/(tabs)/(signer,explorer,converter)/settings/developerRecover.tsx:85]: I would like to introduce zod in this instances

File: apps/mobile/app/(authenticated)/(tabs)/(signer,explorer,converter)/signer/ecash/index.tsx [hunk 8]
```diff
@@ -19,6 +20,17 @@ import { Colors, Sizes } from '@/styles'
...
 
+const ECASH_CARD_FILL_TOP = Colors.gray[905]
+const ECASH_CARD_FILL_BOTTOM = Colors.gray[925]
+const ECASH_CARD_SHEEN_COLORS = [
+  'rgba(255, 255, 255, 0.04)',
+  'rgba(255, 255, 255, 0)',
+  'rgba(255, 255, 255, 0.012)'
+] as const
+
+/** Darker than `muted` for sats / BTC / fiat balance labels */
```
Comment (Psycarlo) [inline apps/mobile/app/(authenticated)/(tabs)/(signer,explorer,converter)/signer/ecash/index.tsx]: Need this comment?

File: apps/mobile/app/(authenticated)/(tabs)/(signer,explorer,converter)/signer/lightning/node/transaction/[txId].tsx [hunk 9]
```diff
@@ -0,0 +1,962 @@
...
+import type {
+  LndInvoice,
+  LndOnchainTransaction,
+  LndPayment
+} from '@/types/lndNodeDashboard'
+import type { LNDGraphNodeInfo } from '@/types/models/LND'
+import { formatFiatPrice, formatNumber } from '@/utils/format'
+import { formatLightningTxTimeAgo } from '@/utils/lndTransactionDisplay'
+
+const PRIVACY_MASK = '••••'
```
Comment (Psycarlo) [inline apps/mobile/app/(authenticated)/(tabs)/(signer,explorer,converter)/signer/lightning/node/transaction/[txId].tsx]: We have so many PRIVACY_MASK vars across the repo xD

File: apps/mobile/app/(authenticated)/(tabs)/(signer,explorer,converter)/signer/lightning/node/transaction/[txId].tsx [hunk 10]
```diff
@@ -0,0 +1,962 @@
...
+type Hop = LndPayment['htlcs'][0]['route']['hops'][0]
+
+function shortPubkey(pk: string): string {
+  if (!pk) {
+    return '—'
+  }
+  return `${pk.slice(0, 8)}…${pk.slice(-6)}`
+}
+
+function HopDiagram({
```
Comment (Psycarlo) [inline apps/mobile/app/(authenticated)/(tabs)/(signer,explorer,converter)/signer/lightning/node/transaction/[txId].tsx:115]: A lot of components being define inside a page...

File: apps/mobile/app/(authenticated)/(tabs)/(signer,explorer,converter)/signer/lightning/node/transaction/[txId].tsx [hunk 11]
```diff
@@ -0,0 +1,962 @@
...
+                      {hop.fee && hop.fee !== '0' && (
+                        <SSText size="xxs" color="muted">
+                          fee {hop.fee} sat / {hop.fee_msat} msat
+                        </SSText>
+                      )}
+                    </>
+                  )}
+                </View>
+              </View>
+
```
Comment (Psycarlo) [inline apps/mobile/app/(authenticated)/(tabs)/(signer,explorer,converter)/signer/lightning/node/transaction/[txId].tsx]: Remove empty line

File: apps/mobile/app/(authenticated)/(tabs)/(signer,explorer,converter)/signer/lightning/node/transaction/[txId].tsx [hunk 12]
```diff
@@ -0,0 +1,962 @@
...
+            items={[
+              [
+                t('lightning.node.txDetail.paymentRequest'),
+                privacyMode ? PRIVACY_MASK : payment.payment_request
+              ]
+            ]}
+          />
+        ) : null}
+      </SSVStack>
+
```
Comment (Psycarlo) [inline apps/mobile/app/(authenticated)/(tabs)/(signer,explorer,converter)/signer/lightning/node/transaction/[txId].tsx]: Remove empty line

File: apps/mobile/app/(authenticated)/(tabs)/(signer,explorer,converter)/signer/lightning/node/transaction/[txId].tsx [hunk 13]
```diff
@@ -0,0 +1,962 @@
...
+                  </SSVStack>
+
+                  {tx.description ? (
+                    <SSText size="sm" color="muted" style={{ marginTop: 10 }}>
+                      {privacyMode ? PRIVACY_MASK : tx.description}
+                    </SSText>
+                  ) : null}
+                </SSVStack>
+              )}
+
```
Comment (Psycarlo) [inline apps/mobile/app/(authenticated)/(tabs)/(signer,explorer,converter)/signer/lightning/node/transaction/[txId].tsx]: Remove

File: apps/mobile/app/(authenticated)/(tabs)/(signer,explorer,converter)/signer/lightning/node/transaction/[txId].tsx [hunk 14]
```diff
@@ -0,0 +1,962 @@
...
+              {rawPayment && paymentHops(rawPayment).length > 0 && (
+                <HopDiagram
+                  hops={paymentHops(rawPayment)}
+                  nodeInfoMap={nodeInfoMap}
+                  privacyMode={privacyMode}
+                />
+              )}
+
```
Comment (Psycarlo) [inline apps/mobile/app/(authenticated)/(tabs)/(signer,explorer,converter)/signer/lightning/node/transaction/[txId].tsx]: Remove empty line

File: apps/mobile/app/(authenticated)/(tabs)/(signer,explorer,converter)/signer/lightning/node/transaction/[txId].tsx [hunk 15]
```diff
@@ -0,0 +1,962 @@
...
+              {rawInvoice && (
+                <InvoiceDetail invoice={rawInvoice} privacyMode={privacyMode} />
+              )}
+
+              {rawOnchain && (
+                <OnchainDetail onchain={rawOnchain} privacyMode={privacyMode} />
+              )}
+
```
Comment (Psycarlo) [inline apps/mobile/app/(authenticated)/(tabs)/(signer,explorer,converter)/signer/lightning/node/transaction/[txId].tsx]: Remove empty line

File: apps/mobile/app/(authenticated)/(tabs)/(signer,explorer,converter)/signer/lightning/node/index.tsx [hunk 16]
```diff
@@ -64,7 +64,10 @@ import {
...
   formatLightningTxTimeAgo,
   getTxDisplayInfo,
-  getTxLightningSendFeeSatString
+  getTxLightningSendFeeSatString,
+  getTxStatusBadge,
+  txDateOptions,
+  type TxStatusBadge
 } from '@/utils/lndTransactionDisplay'
 
 const PRIVACY_MASK = '••••'
```
Comment (Psycarlo) [inline apps/mobile/app/(authenticated)/(tabs)/(signer,explorer,converter)/signer/lightning/node/index.tsx]: Look, another `PRIVACY_MASK`

File: apps/mobile/app/(authenticated)/(tabs)/(signer,explorer,converter)/signer/nostr/account/[npub]/note.tsx [hunk 17]
```diff
@@ -536,16 +574,69 @@ export default function NostrNotePage() {
...
+      const reason = error instanceof Error ? error.message : 'unknown'
+      toast.error(`${t('nostrIdentity.note.bookmarkFailed')}: ${reason}`)
+    } finally {
+      api.disconnect()
+      setBookmarkLoading(false)
+    }
+  }
+
+  function handleBookmarkPress() {
+    Alert.alert(t('nostrIdentity.note.bookmarkTitle'), undefined, [
```
Comment (Psycarlo) [inline apps/mobile/app/(authenticated)/(tabs)/(signer,explorer,converter)/signer/nostr/account/[npub]/note.tsx]: Do we want to use native Alert?
Or our modal?

File: apps/mobile/components/SSNostrFeedTabs.tsx [hunk 18]
```diff
@@ -270,11 +325,107 @@ function SSNostrFeedTabs({
...
+    try {
+      const result = await apiRef.current.fetchBookmarks(npub)
+      if (!result) {
+        setNotes([])
+        return
+      }
+
+      const pubBookmarks = parsePublicBookmarks(result.tags)
+      const nsec = identity?.nsec
+      const privBookmarks = (() => {
```
Comment (Psycarlo) [inline apps/mobile/components/SSNostrFeedTabs.tsx]: avoid IIFE

File: apps/mobile/components/SSPaymentMethodPicker.tsx [hunk 19]
```diff
@@ -1,74 +1,103 @@
...
-  onClose,
-  onSelect,
-  methods,
-  amountSats
-}: SSPaymentMethodPickerProps) {
+function SSPaymentMethodPicker(
+  { onSelect, methods, amountSats }: SSPaymentMethodPickerProps,
+  ref: ForwardedRef<BottomSheetMethods>
+) {
+  const btcPrice = usePriceStore((state) => state.btcPrice)
```
Comment (Psycarlo) [inline apps/mobile/components/SSPaymentMethodPicker.tsx]: Receive data as props is ideal, instead of accessing the store directly

File: apps/mobile/components/SSPaymentMethodPicker.tsx [hunk 20]
```diff
@@ -77,19 +106,17 @@ const styles = StyleSheet.create({
...
+    borderColor: Colors.gray[700],
+    borderRadius: 3,
+    borderWidth: 1,
+    paddingHorizontal: 6,
+    paddingVertical: 2
   }
 })
 
-export default SSPaymentMethodPicker
+export default forwardRef(SSPaymentMethodPicker)
```
Comment (Psycarlo) [inline apps/mobile/components/SSPaymentMethodPicker.tsx]: React 19. `forwardRef` deprecated. Pass ref as regular prop.

File: apps/mobile/constants/nostr.ts [hunk 21]
```diff
@@ -24,7 +24,7 @@ export const RELAY_PROTOCOL_PREFIX = 'wss://'
 /** Fallback color when npub is missing, invalid, or member has no color (e.g. device/member list UI). */
 export const NOSTR_FALLBACK_NPUB_COLOR = '#404040'
 /** Short NDK connect when probing whether at least one relay is reachable. */
-export const NOSTR_RELAY_REACHABILITY_TEST_MS = 5000
+export const NOSTR_RELAY_REACHABILITY_TEST_MS = 10000
 /** Shown when settings privacy mode hides Nostr amounts and identifiers. */
 export const NOSTR_PRIVACY_MASK = '••••'
```
Comment (Psycarlo) [inline apps/mobile/constants/nostr.ts]: We define privacy mask here again

File: apps/mobile/utils/navigation.ts [hunk 22]
```diff
@@ -12,3 +12,52 @@ export function showNavigation(path = '/', depth = 0): boolean {
...
+  '/explorer': '/',
+  '/signer': '/',
+  '/signer/ark/account': '/signer/ark',
+  '/signer/bitcoin': '/',
+  '/signer/bitcoin/account': '/signer/bitcoin/accountList',
+  '/signer/ecash/account': '/signer/ecash',
+  '/signer/nostr/account': '/signer/nostr'
+}
+
+export function getBackPath(pathname: string): string {
```
Comment (Psycarlo) [inline apps/mobile/utils/navigation.ts]: Did you search for native expo-router ways of doing this, instead of writing all this code?

**Review by Psycarlo — CHANGES_REQUESTED** [review]

**Comment by Psycarlo** [comment]
### To summarize

**High**
- 🔴 `useEffect` × 4 added (rule 1) — `_layout.tsx`, `useGetAccountAddress.ts`
- 🔴 20+ `as` casts (rule 5) — worst: double `as unknown as` in `api/ecash.ts`, `api/nostr.ts`, `nostrNoteQuotes.ts`, `recoverBackup.ts`
- 🔴 `performRecoverOverwrite` not transactional — clears stores before restore, swallows errors → corrupted state on partial failure

**Medium**
- 🟠 Hardcoded i18n strings — `SSPaymentMethodPicker` (`'Ark'`/`'ECash'`/`'Lightning'`), `SSCameraModal` (`'Tele'`)
- 🟠 `forwardRef` used — React 19 deprecated, use `ref` prop
- 🟠 `note.tsx` ~1.7k lines, mixed concerns — extract `useNoteBookmarks`, `useZapSort`, `useNip05Status`
- 🟠 `_layout.tsx` animation logic in component — extract `useTabBarAnimation`
- 🟠 `let` × 6 (rule 17), `PaymentMethod` type in component file (rule 8), nested ternary in `SSPaste.tsx` (rule 9)
- 🟠 Manual fetch instead of TanStack Query — `useNostrLandingKind0Sync`, `loadZapReceipts` (rule 23)

**Low**
- 🔵 No tests for new utils: `nostrBookmarks`, `nostrNoteQuotes`, zap sort, recover branches
- 🔵 `transaction/[txId].tsx` 962 new lines: split

**Review by Psycarlo — APPROVED** [review]
